kpi-tracker/src/main.rs

```diff
@@ -0,0 +1,248 @@
+type BlocksTable = HashMap<BlockHash, BlockDetails>;
+
+/// This is intended as a in-memory DB, which follows the same schema as the final DB will follow.
+struct DB {
+    blocks: BlocksTable,
+    accounts: AccountsTable,
+}
+
+async fn account_details(
+    node: &mut Client,
+    block_hash: &BlockHash,
+    account: &AccountAddress,
+) -> anyhow::Result<AccountDetails> {
+    let account_info = node
+        .get_account_info(&AccountIdentifier::Address(*account), block_hash)
+        .await?
+        .response;
+
+    let is_initial =
+        account_info
+            .account_credentials
+            .get(&0.into())
+            .map_or(false, |cdi| match &cdi.value {
+                AccountCredentialWithoutProofs::Initial { .. } => true,
+                AccountCredentialWithoutProofs::Normal { .. } => false,
+            });
+
+    let account_details = AccountDetails {
+        is_initial,
+        block_hash: *block_hash,
+    };
+
+    Ok(account_details)
+}
+async fn accounts_in_block(
+    node: &mut Client,
+    block_hash: &BlockHash,
+) -> anyhow::Result<BTreeMap<AccountAddress, AccountDetails>> {
+    let accounts = node
+        .get_account_list(block_hash)
+        .await
+        .context(format!("Could not get accounts for block: {}", block_hash))?
+        .response;
+
+    let accounts_details_map = accounts
+        .then(|res| {
+            let mut node = node.clone();
+
+            async move {
+                let account = res?;
+                let details = account_details(&mut node, block_hash, &account).await?;
+
+                Ok((account, details))
+            }
+        })
+        .try_fold(BTreeMap::new(), |mut map, (account, details)| async move {
+            map.insert(account, details);
+            Ok(map)
+        })
+        .await;
+
+    accounts_details_map
+}
+
+/// Returns a Map of AccountAddress, AccountDetails pairs not already included in `accounts_table`
+async fn new_accounts_in_block(
+    node: &mut Client,
+    block_hash: &BlockHash,
+    accounts_table: &AccountsTable,
+) -> anyhow::Result<BTreeMap<AccountAddress, AccountDetails>> {
+    let mut transactions = node
+        .get_block_transaction_events(block_hash)
+        .await
+        .context(format!(
+            "Could not get transactions for block: {}",
+            block_hash
+        ))?
+        .response;
+
+    let mut new_accounts = BTreeMap::new();
+
+    while let Some(res) = transactions.next().await {
+        let transaction = res.context("Stream stopped prematurely")?;
+
+        match transaction.details {
+            AccountCreation(act)
+                if !accounts_table
+                    .keys()
+                    .into_iter()
+                    .any(|stored_address| act.address.is_alias(stored_address)) =>
+            {
+                let address = act.address;
+                let account_details = account_details(node, block_hash, &address).await?;
```

Comment:
You should not be querying `get_account_info`, at least not for the data you are getting now. Whether an account is initial or not (see
```

#[derive(Debug, Clone)]
/// Details of an account creation. These transactions are free, and we only
/// ever get a response for them if the account is created, hence no failure
/// cases.
pub struct AccountCreationDetails {
    /// Whether this is an initial or normal account.
    pub credential_type: CredentialType,
    /// Address of the newly created account.
    pub address:         AccountAddress,
    /// Credential registration ID of the first credential.
    pub reg_id:          CredentialRegistrationID,
}
```